### ISSUE TITLE
New TAG for each ruleset file

### ISSUE THREAD

**Issue by theMiddleBlue** [coreruleset/coreruleset#1866]
### Motivation

sometimes I need to exclude a whole CRS ruleset file (included on the main modsecurity.conf) just when occurs a certain condition. For example when 2 different websites share the same modsecurity.conf but need two completely different configurations.

### Proposed solution

Add a tag for each rule on the same ruleset file that could be used to remove them "on the fly". For example:
all rules in REQUEST-912-DOS-PROTECTION.conf have the tag `file-request-dos-protection`
all rules in REQUEST-930-APPLICATION-ATTACK-LFI.conf have the tag `file-request-application-attack-lfi`
etc...

### Alternatives

Or just set a tag based on rules prefix number, something like:
all rules in REQUEST-912-DOS-PROTECTION.conf have the tag `request-912`
all rules in REQUEST-930-APPLICATION-ATTACK-LFI.conf have the tag `request-930`
etc...

if you all agree, I can send a PR

**Comment by dune73** [comment]
Interesting thought, but there are also counter-arguments. We need to talk this through.

**Comment by franbuehler** [comment]
Outcome of chat meeting September: https://github.com/coreruleset/coreruleset/issues/1869#issuecomment-688474359
@theMiddleBlue is answering to @walter about #1866:


> are there downsides of working with the range parameter, e.g. 932000-932999?

I’m not using range in ruleRemoveTargetById because of a compatibility problem with v3 that seems not solved yet as you can see here: https://github.com/SpiderLabs/ModSecurity/issues/2110 (bearbeitet)

**Comment by azurit** [comment]
What are the drawbacks of this? I like this idea too.

**Comment by theMiddleBlue** [comment]
thanks to @airween msc_pyparser module, I've created a script to append, rename or remove tags from rules in file (if someone need it I can push it somewhere):

```sh
python3 rule-ctl.py \
   --config ../../rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf \
   --filter-rule-id '^933.+' \
   --append-tag 'req-933' > tmp/test.conf
```

![image](https://user-images.githubusercontent.com/4454961/116206250-e4996080-a73e-11eb-8574-04c09bb4a0a7.png)

how about adding a short tag like (req|res)-<3 digit id> ?

**Comment by dune73** [comment]
Wow, that's slick. Can we integrate a folder with utilities such as this? Mighty useful, I think.

**Comment by theMiddleBlue** [comment]
Sure! In my test I put the script in util/rule-ctl, how does it sound?
At the moment it has only the following functionalities:

--append-tag
--remove-tag
--rename-tag

but I would like to extend this in order to manage all actions, operators or variables

**Comment by dune73** [comment]
Awesome plan. Let's talk about a good placement at the next project chat (-> add to agenda)

**Comment by dune73** [comment]
Are you guys OK with adding the tags yourself, or do you really think we need one per rule?

Existing drawbacks:
* You can issue a remove rule by ID, even if there is a bug with this in ModSec3. But a fix has been merged for 3.1.0
* We are reluctant to add more tags, since they eat into the alert message space and can lead to tags being eaten and corrupt alert messages

A workaround is apparently adding the tags yourself if you need them.

So if you could live with this solution / workaround, that would be helpful and we could close this issue. 

Please share your opinion and shout if you want to keep this open.

**Comment by azurit** [comment]
It's not a killer feature for me, so..

**Comment by theMiddleBlue** [comment]
if the v3 bug has been fixed on 3.1.0, this could be closed

**Comment by dune73** [comment]
Thank you very much for your prompt response. Much appreciated you can live without these new tags. Closing this.
